### PR TITLE
Update the main TiDB version to v5.0 in e2e

### LINKED ISSUE #3877: Update the main TiDB version to v5.0 in e2e
## Feature Request

**Is your feature request related to a problem? Please describe:**

**Describe the feature you'd like:**

TiDB v5.0 will be released on 03/31/2021, we should update the e2e to focus on the new GA version.
In addition, v3.0 and v4.0 should also be covered in the basic case.

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by handlerww**
### What problem does this PR solve?

Closes #3877
### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

**Review by july2993 — APPROVED** [review]

File: tests/e2e/tidbcluster/tidbcluster.go [hunk 1]
```diff
@@ -1537,7 +1537,59 @@ var _ = ginkgo.Describe("TiDBCluster", func() {
...
+			tidbSts, err := stsGetter.StatefulSets(ns).Get(tidbMemberName, metav1.GetOptions{})
+			framework.ExpectNoError(err, "failed to get StatefulSet %s/%s", ns, tidbMemberName)
+			tidbImage := fmt.Sprintf("pingcap/tidb:%s", componentVersion)
+			// the 0th container for tidb pod is slowlog, which runs busybox
+			framework.ExpectEqual(tidbSts.Spec.Template.Spec.Containers[1].Image, tidbImage, "tidb sts image should be %q", tidbImage)
+		})
+
+		ginkgo.It("for tc and components version upgrade from TiDB V4 to TiDB V5", func() {
+			ginkgo.By("Deploy initial tc")
+			tc := fixture.GetTidbCluster(ns, "upgrade-version-v4-to-v5", utilimage.TiDBV4)
```
Comment (DanielZhangQD) [inline tests/e2e/tidbcluster/tidbcluster.go:1592]: No need to use `local-storage` and `Retain`.

File: tests/e2e/tidbcluster/tidbcluster.go [hunk 2]
```diff
@@ -1537,7 +1537,7 @@ var _ = ginkgo.Describe("TiDBCluster", func() {
 	ginkgo.Context("upgrade should work correctly", func() {
```
[the remark this answers is not in the document]
Comment (handlerww) [reply tests/e2e/tidbcluster/tidbcluster.go:1537]: I'll fix this in another PR.

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Pull request merged**